app/views/pages/get-started/contact-us-tutorial/using-events.liquid

```diff
@@ -0,0 +1,396 @@
+
+{% raw %}
+```liquid
+assign c = '{ "errors": {}, "valid": true }' | parse_json
+```
+{% endraw %}
+
+This line initializes a JSON object `c` with an empty `errors` hash and sets `valid` to `true`.
+
+{% raw %}
+```liquid
+function c = 'modules/core/validations/presence', c: c, object: event, field_name: 'id'
+```
+{% endraw %}
+
+To ensure the contact ID is provided and valid, we use validation checks. The `presence` validation confirms that the contact ID is included in the event data.
+
+{% raw %}
+```liquid
+return c
+```
+{% endraw %}
+
+The function returns the validation result, including any errors and the `valid` status.
+
+## Publish the Event
+
+After defining and validating the event, the next step is to publish it. Publishing an event means notifying the system that something significant has happened, like the submission of a contact form. This step helps in tracking actions and triggers any required follow-up processes, such as sending confirmation emails or updating statuses.
+
+Copy the following code snippet from the core module documentation for [publishing an event](https://github.com/Platform-OS/pos-module-core?tab=readme-ov-file#publishing-event):
+
+{% raw %}
+```liquid
+  assign object = null | hash_merge: foo_id: "12345"
+  function _ = 'modules/core/commands/events/publish', type: 'something_happened', object: object
+```
+{% endraw %}
+
+Paste the copied code into your `app/lib/commands/contacts/create.liquid` file. This file already has the commands for building, checking, and executing the contact creation process. Integrate the publishing step as shown below:
+
+#### app/lib/commands/contacts/create.liquid
+
+{% raw %}
+```liquid
+{% liquid
+ function object = 'commands/contacts/create/build', object: object
+ function object = 'commands/contacts/create/check', object: object
+
+ if object.valid
+   function object = 'commands/contacts/create/execute', object: object
+   function _ = 'modules/core/commands/events/publish', type: 'contact_created', object: object
+ endif
+
+ return object
+%}
+```
+{% endraw %}
+
+## Understanding the publish function
+
+The `publish` function in platformOS serves to create and store events, allowing you to track and respond to significant actions in your application.
+
+It operates in the background, ensuring it doesn't block the main execution flow. This means that while your application continues to run, the event is logged in parallel.
+
+Following the command pattern, publish calls the `events/create` function defined in the core module. This modular approach allows for clear, maintainable code.
+
+The event logs the timestamp and the object provided (in this case, the contact ID). This logging mechanism helps in tracking the sequence of actions and debugging issues.
+
+Instead of creating a database record, the event is stored in an activity stream. This stream captures the payload (contact ID) and the event name (`contact_created`).
```

Comment:
What does that actually mean. What are the benefits, what are the differences. Is it cheaper/faster?

Reply:
we might want to loop in @godot to answer this comment ;)